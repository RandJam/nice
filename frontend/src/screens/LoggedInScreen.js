import React, {Component} from 'react';
import {Text, View, TouchableOpacity } from 'react-native';
const util = require('util')

export default class LoggedIn extends Component {

  handleButtonScreenPress = () => {
     this.props.navigation.navigate("ButtonScreen");
  }

  render(){
      return (
      <View>
        <Text> LoggedIn </Text>
        <TouchableOpacity onPress={this.handleButtonScreenPress}>
          <Text> Go to button </Text>
        </TouchableOpacity>
      </View>
    )

  }
}
