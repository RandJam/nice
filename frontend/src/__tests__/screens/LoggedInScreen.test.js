import React from "react";
import {shallow} from 'enzyme';
import LoggedInScreen from "../../screens/LoggedInScreen";
import renderer from "react-test-renderer";

describe ("LoggedInScreen", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<LoggedInScreen />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });

  it('calls navigate on pressing go to button', () => {
    const fakeNavigation = {navigate: jest.fn()};

    let loggedIn = renderer.create(<LoggedInScreen
      navigation={fakeNavigation}
      />
    ).getInstance();

    loggedIn.handleButtonScreenPress();
    expect(fakeNavigation.navigate).toHaveBeenCalledTimes(1);
  })
})
