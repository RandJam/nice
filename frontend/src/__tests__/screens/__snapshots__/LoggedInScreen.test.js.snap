// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInScreen renders without crashing 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     LoggedIn 
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       Go to button 
    </Text>
  </View>
</View>
`;
